apps/charterafrica/src/pages/index.page.js

```diff
@@ -42,6 +42,35 @@ export async function getStaticProps({ defaultLocale, locale, locales }) {
     fallbackLocale: defaultLocale,
   });
 
+  const { docs: homePage } = await payload.findPage("home", "pages", {
+    locale,
+    fallbackLocale: defaultLocale,
+  });
+
+  const { blocks } = homePage[0] ?? {
+    blocks: [],
+  };
+
+  const spotlightItems = blocks
+    .filter((block) => block.blockType === "spotlight")[0]
+    ?.items.map((spotlight) => {
+      const { item, ...rest } = spotlight;
+      const formattedItem = {
+        ...item,
+        image: {
+          src: item.image.url,
+          alt: item.image.alt,
+        },
+        date: new Date(item.date).toLocaleDateString(),
+      };
+      return {
+        ...rest,
+        item: {
+          ...formattedItem,
+        },
+      };
```

Comment:
Does the `link()` field return `href` directly?

Reply:
Yes. `href` is returned directly

Reply:
I odn't see href on link @kelvinkipruto . 👇🏽 is the structure I'm getting back.
```json
  "blocks": [
    {
      "items": [
        {
          "category": "UPCOMING BLOG",
          "item": {
            "title": "Title",
            "image": {
              "id": "638999dea60f310b0f5299ce",
              "alt": "The Charter Project Africa",
              "prefix": "media",
              "filename": "image.jpg",
              "mimeType": "image/jpeg",
              "filesize": 178719,
              "width": 1170,
              "height": 622,
              "createdAt": "2022-12-02T06:23:26.825Z",
              "updatedAt": "2022-12-02T06:23:26.825Z",
              "url": "http://localhost:3003/media/image.jpg"
            },
            "topic": "TOPIC",
            "excerpt": "Excerpt",
            "date": "2023-01-15T21:00:00.000Z",
            "type": "reference",
            "reference": {
              "value": {
                "id": "63887cf05bc566facccee049",
                "title": "Home",
                "slug": "index",
                "prefix": "media",
                "filename": "CharterAfrica-logoboard 1.png",
                "mimeType": "image/png",
                "filesize": 33157,
                "width": 512,
                "height": 573,
                "createdAt": "2022-12-01T10:07:44.251Z",
                "updatedAt": "2023-01-16T13:44:16.210Z",
                "meta": {
                  "title": "The Charter Project Africa",
                  "description": "A pan-African project focused on the commitments contained in the African Charter for Democracy, Elections and Governance (ACDEG)",
                  "image": "638999dea60f310b0f5299ce"
                },
                "blocks": [
                  {
                    "items": [
                      {
                        "category": "UPCOMING BLOG",
                        "item": {
                          "title": "Title",
                          "image": "638999dea60f310b0f5299ce",
                          "topic": "TOPIC",
                          "excerpt": "Excerpt",
                          "date": "2023-01-15T21:00:00.000Z",
                          "type": "reference",
                          "reference": {
                            "value": "63887cf05bc566facccee049",
                            "relationTo": "pages"
                          },
                          "label": "Label"
                        },
                        "id": "63c5547c5514eef9f8838ce5"
                      }
                    ],
                    "id": "63c5547a5514eef9f8838ce4",
                    "blockType": "spotlight"
                  }
                ]
              },
              "relationTo": "pages"
            },
            "label": "Label"
          },
          "id": "63c5547c5514eef9f8838ce5"
        }
      ],
      "id": "63c5547a5514eef9f8838ce4",
      "blockType": "spotlight"
    }
  ]
```

Reply:
There is no link when the link type is `Internal Link`, however it is there when it is `Custom URL`. 
How should we handle references to internal links?. For example in your above response, the reference does not return a `href` we can use.

Reply:
This is what I was asking you @kelvinkipruto ... but since we're pressed for time, lets just focus on external links for now.

👇🏽  is the structure
```json
  "blocks": [
    {
      "items": [
        {
          "category": "UPCOMING BLOG",
          "item": {
            "title": "Title",
            "image": {
              "id": "638999dea60f310b0f5299ce",
              "alt": "The Charter Project Africa",
              "prefix": "media",
              "filename": "image.jpg",
              "mimeType": "image/jpeg",
              "filesize": 178719,
              "width": 1170,
              "height": 622,
              "createdAt": "2022-12-02T06:23:26.825Z",
              "updatedAt": "2022-12-02T06:23:26.825Z",
              "url": "http://localhost:3003/media/image.jpg"
            },
            "topic": "TOPIC",
            "excerpt": "Excerpt",
            "date": "2023-01-15T21:00:00.000Z",
            "type": "custom",
            "reference": {
              "value": {
                "id": "63887cf05bc566facccee049",
                "title": "Home",
                "slug": "index",
                "prefix": "media",
                "filename": "CharterAfrica-logoboard 1.png",
                "mimeType": "image/png",
                "filesize": 33157,
                "width": 512,
                "height": 573,
                "createdAt": "2022-12-01T10:07:44.251Z",
                "updatedAt": "2023-01-16T14:29:39.540Z",
                "meta": {
                  "title": "The Charter Project Africa",
                  "description": "A pan-African project focused on the commitments contained in the African Charter for Democracy, Elections and Governance (ACDEG)",
                  "image": "638999dea60f310b0f5299ce"
                },
                "blocks": [
                  {
                    "items": [
                      {
                        "category": "UPCOMING BLOG",
                        "item": {
                          "title": "Title",
                          "image": "638999dea60f310b0f5299ce",
                          "topic": "TOPIC",
                          "excerpt": "Excerpt",
                          "date": "2023-01-15T21:00:00.000Z",
                          "type": "custom",
                          "reference": {
                            "value": "63887cf05bc566facccee049",
                            "relationTo": "pages"
                          },
                          "href": "https://codeforafrica.org",
                          "label": "Label"
                        },
                        "id": "63c5547c5514eef9f8838ce5"
                      }
                    ],
                    "id": "63c5547a5514eef9f8838ce4",
                    "blockType": "spotlight"
                  }
                ]
              },
              "relationTo": "pages"
            },
            "href": "https://codeforafrica.org",
            "label": "Label"
          },
          "id": "63c5547c5514eef9f8838ce5"
        }
      ],
      "id": "63c5547a5514eef9f8838ce4",
      "blockType": "spotlight"
    }
  ]
```

1. There is blocks with items. This shouldn't happen (unless I'm missing something).
2. Looking at hard-coded data, the expected structure for `link` is `{ "link": {  "href": "https://....." } }`. I think `link()` is missing `name` at least and _may be_ should be in a `group.`

Reply:
@kilemensi Yes, giving it a name solves the issue